### PR TITLE
AndroidX Support

### PR DESCRIPTION
**Pull request by mrbrentkelly**
This PR updates the android project (and example android project) to support AndroidX. 

React Native will support AndroidX once 0.60 is released, so it's important that third party libraries also make the switch.

https://github.com/facebook/react-native/releases/tag/v0.60.0-rc.1
https://github.com/facebook/react-native/issues/23112

**Note: I would advise NOT merging this until RN0.60 has been released.**

### CODE REVIEW

File: android/gradle.properties
Comment (se-bastiaan) [inline android/gradle.properties]: This is not necessary since there are no other dependencies that this library depends on.

File: example/android/gradle.properties
Comment (se-bastiaan) [inline example/android/gradle.properties]: This is not necessary since there are no other dependencies that this library depends on.

**Review by se-bastiaan — CHANGES_REQUESTED** [review]

**Comment by cooperka** [comment]
Much appreciated @mrbrentkelly, I'll merge this once v0.60 is released (feel free to remind me when that happens as I'm not highly active with RN these days).

**Comment by bhaskarGyan** [comment]
Hi @cooperka , v0.60 is released, Can you please merge

File: example/android/build.gradle
Comment (se-bastiaan) [inline example/android/build.gradle:35]: Lines 23-25 already contain this maven repository. Those lines can be replaced by these.

**Review by se-bastiaan — CHANGES_REQUESTED** [review]

**Comment by bhaskarGyan** [comment]
Hi @se-bastiaan , @cooperka Should I open a new PR with above changes?
Looks like @mrbrentkelly is busy.

File: example/android/gradle.properties
Comment (mrbrentkelly) [reply example/android/gradle.properties]: @se-bastiaan if I remove this line then the sample app fails to build. Maybe because it's running an older version of React Native (0.56.0), which will have Android dependencies that need run through the jetifier? (The fix in that case would be to upgrade react-native-snackbar to RN 0.60, or run with jetifier enabled and upgrade in a subsequent PR)

File: example/android/build.gradle
Comment (mrbrentkelly) [reply example/android/build.gradle:35]: Updated to match a fresh React Native project 👍

File: android/gradle.properties
Comment (mrbrentkelly) [reply android/gradle.properties]: Removed 👍

File: example/android/gradle.properties
Comment (se-bastiaan) [reply example/android/gradle.properties]: I believe that would be the reason indeed. I'm not sure it is wise to give an example with <0.60 if the update to the package itself will possibly break applications that have not updated to RN 0.60. At least if they do not use jetifier themselves. I'm seeing a lot of different approaches here, I'm not even sure what [the proposed option by the react-native maintainers is exactly](https://github.com/react-native-community/discussions-and-proposals/issues/129#issuecomment-507010053). In the case of this package I think it is a valid approach to drop support for pre-0.60 and go to version 2.0.0, since it heavily depends on the support library/material library to provide the UI. In that case the example would also need to be 0.60.

Comment (se-bastiaan) [reply example/android/gradle.properties]: Okay so it seems that a special jetifier tool for react-native is the recommended path to updating: https://github.com/mikehardy/jetifier

Comment (mrbrentkelly) [reply example/android/gradle.properties]: Agreed, I think we should go to version 2.0.0 👍. 

I have went ahead and updated the example app to React Native 0.60.3. @se-bastiaan, could you give this PR another code review 🙏?

cc @cooperka

**Comment by cooperka** [comment]
Fantastic work here, thank you all 🎉

Tested and verified on Android and iOS. Will bump major version to 2.0. Much appreciated.

**Pull request merged**
